### PR TITLE
implemented circle.radius setter

### LINKED ISSUE #224: <geo-circle>.radius is not implemented
We can't currently set the radius of circles from script.

### PR DESCRIPTION
**Pull request by strobelm**
Hi, i quickly implemented the circle.radius setter. 

I'm not quite sure if the direct manipulation of stateIn is the right way to do that therefor i would really appreciate comments on that. This should fix #224 and #254.

### CODE REVIEW

**Comment by kranich** [comment]
@strobel: You must not directly update `stateIn`. This is likely to break tracing if there is a point on the circle that would be moved across a great distance when the radius changes abruptly.
--
Stefan Kranich
Technische Universität München -- Zentrum Mathematik
Lehrstuhl für Geometrie und Visualisierung (M10)
Boltzmannstr. 3, 85747 Garching bei München

**Comment by strobelm** [comment]
@kranich: okay, is there a tracing-safe way to achieve this?

Am 07.03.2016 um 15:10 schrieb Stefan Kranich:
> @strobel: You must not directly update `stateIn`. This is likely to
> break tracing if there is a point on the circle that would be moved
> across a great distance when the radius changes abruptly.
> --
> Stefan Kranich
> Technische Universität München -- Zentrum Mathematik
> Lehrstuhl für Geometrie und Visualisierung (M10)
> Boltzmannstr. 3, 85747 Garching bei München
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CindyJS/CindyJS/pull/293#issuecomment-193260710>.
>

-- 
Michael Strobel, M.Sc.

Technische Universität München -- Zentrum Mathematik
Lehrstuhl für Geometrie und Visualisierung (M10)
Boltzmannstr. 3, 85747 Garching bei München 
https://www-m10.ma.tum.de/bin/view/Lehrstuhl/MichaelStrobel

**Comment by kranich** [comment]
@strobel Please proceed analogously to `movepointscr`.

Am 7. März 2016 15:12:49 MEZ, schrieb Michael Strobel <notifications@github.com>:
>@kranich: okay, is there a tracing-safe way to achieve this?
>
>Am 07.03.2016 um 15:10 schrieb Stefan Kranich:
>> @strobel: You must not directly update `stateIn`. This is likely to
>> break tracing if there is a point on the circle that would be moved
>> across a great distance when the radius changes abruptly.
>> --
>> Stefan Kranich
>> Technische Universität München -- Zentrum Mathematik
>> Lehrstuhl für Geometrie und Visualisierung (M10)
>> Boltzmannstr. 3, 85747 Garching bei München
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/CindyJS/CindyJS/pull/293#issuecomment-193260710>.
>>
>
>-- 
>Michael Strobel, M.Sc.
>
>Technische Universität München -- Zentrum Mathematik
>Lehrstuhl für Geometrie und Visualisierung (M10)
>Boltzmannstr. 3, 85747 Garching bei München 
>https://www-m10.ma.tum.de/bin/view/Lehrstuhl/MichaelStrobel
>
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/CindyJS/CindyJS/pull/293#issuecomment-193262268

--
Stefan Kranich
Technische Universität München -- Zentrum Mathematik
Lehrstuhl für Geometrie und Visualisierung (M10)
Boltzmannstr. 3, 85747 Garching bei München

**Comment by strobelm** [comment]
So here's the update, again comments are very welcome. I'm now passing directly the desired parameter to the traceMover.

This is not a general solution for tracing scripts, we should consider a rewrite in the near future.

**Comment by gagern** [comment]
In my opinion, the `type` argument of `traceMover` is free to be used in any way that makes sense for both its caller and the geometric primitive being traced. So in the case of `CircleMr` one may simply introduce a new `type` called `"radius"` and associate with that the meaning that `pos`, `radius` and `parameter` are all the same thing. So `getParamForInput(someCircleMr, r, "radius")` should just retern `r`. That's what I did in the commit @strobelm just merged.

**Comment by strobelm** [comment]
Hi, i like the solution. Perhaps @kranich could have a final look.

**Comment by gagern** [comment]
To formulate this in a way [GitHub understands](https://help.github.com/articles/closing-issues-via-commit-messages/): this change also fixes #254.

File: src/js/libgeo/Tracing.js [hunk 1]
```diff
@@ -140,7 +140,7 @@ function movepointscr(mover, pos, type) {
 /*
  * traceMover moves mover from current param to param for pos along a complex detour.
  */
-function traceMover(mover, pos, type) {
+function traceMover(mover, pos, type, knownTargetParam) {
```
Comment (kranich) [inline src/js/libgeo/Tracing.js]: @strobelm The changes in [Tracing.js](https://github.com/strobelm/CindyJS/blob/CircleRadius/src/js/libgeo/Tracing.js) are obsolete. Please remove all occurences of `knownTargetParam` in `traceMover`.

File: src/js/libgeo/Tracing.js [hunk 2]
```diff
@@ -155,7 +155,7 @@ function traceMover(mover, pos, type) {
     stateInIdx = mover.stateIdx;
     var originParam = opMover.getParamFromState(mover);
     stateInIdx = stateOutIdx = mover.stateIdx;
-    var targetParam = opMover.getParamForInput(mover, pos, type);
+    var targetParam = knownTargetParam || opMover.getParamForInput(mover, pos, type);
```
Comment (kranich) [inline src/js/libgeo/Tracing.js]: See above.

**Comment by strobelm** [comment]
@kranich thanks for the pull request, i already pushed a similar one some seconds ago. Perhaps we should squash all the commits to one?

**Comment by kranich** [comment]
>@kranich thanks for the pull request, i already pushed a similar one 
>some seconds ago. Perhaps we should squash all the commits to one?

OK, please squash.

-- 
Stefan Kranich
Technische Universität München -- Zentrum Mathematik
Lehrstuhl für Geometrie und Visualisierung (M10)
Boltzmannstr. 3, 85747 Garching bei München

**Comment by strobelm** [comment]
On 03/07/2016 09:07 PM, Stefan Kranich wrote:
> >@kranich thanks for the pull request, i already pushed a similar one
> >some seconds ago. Perhaps we should squash all the commits to one?
>
> OK, please squash.
>
> -- 
> Stefan Kranich
> Technische Universität München -- Zentrum Mathematik
> Lehrstuhl für Geometrie und Visualisierung (M10)
> Boltzmannstr. 3, 85747 Garching bei München
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CindyJS/CindyJS/pull/293#issuecomment-193427140>.
>

Done, thanks very much @gagern and @kranich!

-- 
Michael Strobel, M.Sc.

Technische Universität München -- Zentrum Mathematik
Lehrstuhl für Geometrie und Visualisierung (M10)
Boltzmannstr. 3, 85747 Garching bei München 
https://www-m10.ma.tum.de/bin/view/Lehrstuhl/MichaelStrobel

**Comment by kranich** [comment]
@gagern, @strobelm Thank you.

**Comment by kranich** [comment]
@gagern GitHub did not close #254 automatically. Maybe the keywords only work in the description but not in comments.

**Pull request merged**